mtags/src/main/scala-2/scala/meta/internal/pc/InferredMethodProvider.scala

```diff
@@ -0,0 +1,496 @@
+            // this gives me the position of the class
+            typedTreeAt(containerClass.symbol.pos) match {
+              // class case
+              case ClassDef(
+                    _,
+                    _,
+                    _,
+                    template
+                  ) =>
+                makeClassMethodEdits(template)
+              // trait case
+              case ModuleDef(
+                    _,
+                    _,
+                    template
+                  ) =>
+                makeClassMethodEdits(template)
+              case _ =>
+                // object Y {}
+                // Y.nonExistent(1,2,3)
+                typedTreeAt(containerSymbol.symbol.pos) match {
+                  // class case
+                  case ClassDef(
+                        _,
+                        _,
+                        _,
+                        template
+                      ) =>
+                    makeClassMethodEdits(template)
+                  // trait case
+                  case ModuleDef(
+                        _,
+                        _,
+                        template
+                      ) =>
+                    makeClassMethodEdits(template)
+                  case _ =>
+                    Nil
+                }
+            }
+          case _ =>
+            Nil
+        }
+      } else
+        Nil
+    }
+    val uriTextEditPairs = typedTree match {
+      case errorMethod: Ident if errorMethod.isErroneous =>
+        lastVisitedParentTrees match {
+          /**
+           * Works for apply with unknown name:
+           * ```scala
+           * method(nonExistent(param))
+           * ```
+           */
+          case _ :: (nonExistent: Apply) :: Apply(
+                Ident(containingMethod),
+                arguments
+              ) :: _ if nonExistent.isErrorTyped =>
+            makeEditsForApplyWithUnknownName(
+              arguments,
+              containingMethod,
+              errorMethod,
+              nonExistent
+            )
+          // nonExistent(param1, param2)
+          // val a: Int = nonExistent(param1, param2)
+          case (_: Ident) :: Apply(
+                containing @ Ident(
+                  nonExistent
+                ),
+                arguments
+              ) :: _ if containing.isErrorTyped =>
+            signature(
+              name = nonExistent.toString(),
+              paramsString = argumentsString(arguments).getOrElse(""),
+              retType = None,
+              postProcess = identity,
+              position = None
+            )
+          // List(1, 2, 3).map(nonExistent)
+          // List((1, 2)).map{case (x,y) => otherMethod(x,y)}
+          case (_: Ident) :: Apply(
+                Ident(containingMethod),
+                arguments
+              ) :: _ =>
+            makeEditsForListApply(arguments, containingMethod, errorMethod)
+          // List(1,2,3).map(myFn)
```

Comment:
This and the one below does not work unfortunately, it's not really possible to get a type of List(1,2,3).map the current implementation will just produce:

def otherMethod(arg0: Int, arg1: Int, arg2: Int) = ???

which is not valid. We can't even know how many parameters it will have. And also this might be a much more complex expression.